src/groovy/groovy/com/palantir/gradle/gitversion/BuildScanPluginInterop.groovy

```diff
@@ -21,24 +21,36 @@ import org.gradle.api.Project
 
 class BuildScanPluginInterop {
     static void addBuildScanCustomValues(Project rootProject, Supplier<Map<String, String>> customValues) {
-        // After evaluate because while we can detect the <5.x com.gradle.build-scan project on the root project,
-        // there is no way to detect the >6.x com.gradle.enterprise settings plugins using withPlugin
-        rootProject.afterEvaluate {
-            def gradleEnterpriseExtension = rootProject.extensions.findByName('gradleEnterprise')
+        // Fix #353: Detect when the root project is already evaluated, thus anything is afterEvaluate
+        // and we can just execute immediately. This is due to afterEvaluate changes in Gradle 7.
+        //
+        // Note: We cannot use hasCompleted() because we need to support gradle 5
```

Comment:
> we need to support gradle 5

Why?

Reply:
For no other reason that I did not want to take away behavior as I suspected that would require a major version bump. The previous comments seemed to imply that having this work on Gradle 5 was something to be desired? IDC TBH